Testcases.dfy

```diff
@@ -0,0 +1,43 @@
+// This file shows the test cases for Sum, mean and variance
+
+include "StatisticsLibrary.dfy"
+
+module TestModule {
+  import opened StatisticsLibrary
+
+  // Test cases for the StatisticsLibrary functions
```

Comment:
These are great!

For when you move this over to an actual Dafny standard libraries PR, you can replace all the `assert`s with `AssertAndExpect` (e.g. https://github.com/dafny-lang/dafny/blob/80a1382d7719860ec39c41c9f2fdd089657152ea/Source/DafnyStandardLibraries/examples/CollectionsExamples.dfy). That way you ensure the assertions are both statically provable, and actually true at runtime. Doing that alone will probably ensure you have enough execution coverage in your tests.